lfs/batcher.go

```diff
@@ -51,16 +55,22 @@ func (b *Batcher) acceptInput() {
 	exit := false
 
 	for {
-		batch := b.newBatch()
+		batch := make([]Transferable, 0, b.batchSize)
 	Loop:
 		for len(batch) < b.batchSize {
-			t, ok := <-b.input
-			if !ok {
-				exit = true // input channel was closed by Exit()
-				break Loop
+			select {
+			case t, ok := <-b.input:
+				if ok {
+					batch = append(batch, t)
+				} else {
+					exit = true // input channel was closed by Exit()
+					break Loop
+				}
+			case <-time.After(time.Millisecond * 300):
```

Comment:
Not quite. For retrying, the transfer queue was changed slightly. Previously, you'd `Add()` all your items, then `Wait()` would start up everything, but it would call batcher's `Exit()` to ensure any remaining items were flushed from the batcher. The side effect of this is that once you called `Wait()` on the transfer queue, you can't `Add()` anything to it. The timeout ensures batches of less than 100 are flushed implicitly. e.g. if the caller is blocked on `Next()` and there are 50 items waiting to be batched, the caller will get them after 300ms. Doing this means we can just use `Add()` to retry an object.